alicloud/resource_alicloud_cms_alarm.go

```diff
@@ -299,6 +317,15 @@ func resourceAlicloudCmsAlarmUpdate(d *schema.ResourceData, meta interface{}) er
 		update = true
 		request.NotifyType = requests.NewInteger(d.Get("notify_type").(int))
 	}
+	if d.HasChange("webhook") {
+		update = true
+
+		if webhook, ok := d.GetOk("webhook"); ok && webhook.(string) != "" {
+			request.Webhook = cmsService.BuildJsonWebhook(webhook.(string))
+		} else {
+			request.Webhook = ""
+		}
```

Comment:
I think the line 323-327 can be merged to `request.Webhook = cmsService.BuildJsonWebhook(d.Get("webhook").(string))`and you can check whether `webhook` is empty in the method `BuildJsonWebhook`

Reply:
done